sorting/count_sort/count_sort.py

```diff
@@ -0,0 +1,23 @@
+import math
+def counting_sort(L):
+    largest = -math.inf
+    n = len(L)
+    for i in range(n):

```

Comment:
1. Edge case: If the input is an empty list ( [ ] ), the code will tend to break. Handle this case !
2. Edge case: If there are negative numbers, the code can misbehave as you have not handled negative numbers as an edge case.
3. Edge case: If the range of numbers in the list is much larger than the number of elements in L, the algorithm may not be space- and time-efficient. Example: [1, 1000000].....The range size is 1000000 - 1 + 1 = 1000000. The frequency array will have 1,000,000 entries, but only two of those entries will be used. This results in high space complexity (O(range_size)), even though the time complexity is linear (O(n + range_size)).

Reply:
Hey! I implemented and handled the first 2 edge cases. please check it out. The 3rd edge case that you described is the real disadvantage of counting sort and it is suitable for small-range data like student marks which are in the range(1 -100) and are well known to us and obvious. For the 3rd edge case, we can go for an alternate sorting method.

Reply:
Cool...edge cases handled

Reply:
Can you acknowledge if this is a code from GPT ? If not, we can have a meet to discuss upon

Reply:
Hi @FirefoxSRV. For the second edge case, I referred to a research paper on the net that implemented this similar shifting logic to handle negative numbers, which I did not understand properly.
I wrote the previous code by myself and to handle edge case number 2 easily, I referred to this website
 https://yuminlee2.medium.com/counting-sort-algorithm-392560fe570e.I implemented the same logic Please do check it out.

Reply:
Awesome @Leela0o5.
If you can add the research paper link to the code on top, it'll be great!

Reply:
Good one @Leela0o5 !